content/how-to-use-supertokens-custom-ui-with-vuejs/index.md

```diff
@@ -0,0 +1,608 @@
+
+## Architecture
+
+SuperTokens is built out of three components:
+
+- Frontend SDK
+- Backend SDK
+- Core microservice that talks to a database.
+
+We’ll build our own login, signup and reset password forms UI. Then we’ll use the supertokens-web-js SDK in the Vue app to make these forms functional by invoking the relevant functions for each action. These functions will interact with the SuperTokens backend.
+
+For the backend we’ll use the NodeJS SDK provided by SuperTokens. The APIs exposed by this SDK will further talk to the SuperTokens core microservice to read/write information to the database.
+
+The SuperTokens core service can be either self hosted (and connected to your own db), or be hosted by the team behind SuperTokens (sign up on[ supertokens.com](https://supertokens.com/)).
+
+## Frontend
+
+Start by creating a new Vue app:
+
+```shell
+npm init vue@latest
+```
+
+We'll enable Vue Router and Typescript for the project. Chose yes for them in the prompt:
+
+```shell
+✔ Project name: … <your-project-name>
+...
+✔ Add TypeScript? … Yes
+✔ Add Vue Router for Single Page Application development? … Yes
+...
+
+Scaffolding project in ./<your-project-name>...
+Done.
+```
+
+Once that’s done, head inside the project and install the following dependencies:
+
+```sh
+npm i supertokens-node supertokens-web-js cors dotenv express npm-run-all
+```
+
+The `supertokens-web-js` library will be used on the frontend, to add authentication and reset password functionality to our custom UI and the supertokens-node library will be used on the backend to expose the auth API routes, which the frontend calls.
+
+
+
+### **Step 1) Call** **`supertokens-webjs`** **`init`** **function**
+
+We'll initialize the supertokens-webjs SDK in our Vue app's root file, ie `/src/main.ts` :  
+
+```typescript
+...
+import ThirdPartyEmailPassword from "supertokens-web-js/recipe/thirdpartyemailpassword";
+import Session from "supertokens-web-js/recipe/session";
+
+...
+
+
+SuperTokens.init({
+    appInfo: {
+        appName: "SuperTokens Vue ThirdPartyEmailPassword Example",
+        apiDomain: "http://localhost:3001",
+    },
+    recipeList: [ThirdPartyEmailPassword.init(), Session.init()],
+});
+
+...
+
+```
+
+In the above code, the `init` function initializes the SuperTokens Web SDK in our Vue application. Since it’s called in the root of our application, it provides session management across the entire application. It also indicates the type of authentication we want to use - in our case, it’s social login + email password (ThirdPartyEmailPassword recipe). 
```

Comment:
```suggestion
In the above code, the `init` function initializes `supertokens-web-js` on the frontend. We call this function in the root file of our application, so that we can use the session management feature across the entire application. It also indicates the type of authentication we want to use - in our case, it's social login + email password (`ThirdPartyEmailPassword` recipe). 
```